lib/combine_pdf/renderer.rb

```diff
@@ -107,7 +107,7 @@ def format_hash_to_pdf(object)
         end
       end
       # remove extra page references.
-      object[:Contents].delete(is_reference_only: true, referenced_object: { indirect_reference_id: 0, raw_stream_content: '' }) if object[:Type] == :Page
+      object[:Contents].delete(is_reference_only: true, referenced_object: { indirect_reference_id: 0, raw_stream_content: '' }) if object[:Type] == :Page && object[:Contents]
```

Comment:
I only use it to combine simple PDF files.

In the way you described it, `&& object[:Contents].is_a?(Array)` is a better fix.